src/_pytest/capture.py

```diff
@@ -123,9 +123,9 @@ def _py36_windowsconsoleio_workaround(stream: TextIO) -> None:
         return
 
     buffered = hasattr(stream.buffer, "raw")
-    raw_stdout = stream.buffer.raw if buffered else stream.buffer
+    raw_stdout = stream.buffer.raw if buffered else stream.buffer  # type: ignore[attr-defined]
```

Comment:
@bluetech for some reason I had to add those `type: ignore` here in `capture.py` even though the original patch doesn't touch this file... any thoughts?

Reply:
It's because mypy is platform-dependent. By default the platform is `sys.platform` which is linux on CI (and me), but these errors only happen on Windows:

```console
$ mypy --platform=win32 src/**/*.py testing/**/*.py
src/_pytest/capture.py:126: error: "BinaryIO" has no attribute "raw"  [attr-defined]
src/_pytest/capture.py:128: error: Module has no attribute "_WindowsConsoleIO"  [attr-defined]
src/_pytest/capture.py:138: error: Argument 1 to "TextIOWrapper" has incompatible type "IO[Any]"; expected "BinaryIO"  [arg-type]
```

After this is merged, we should probably change CI somehow to check win32 platform as well.